windows/sound.cpp

```diff
@@ -615,243 +776,107 @@ void CSound::bufferSwitch ( long index, ASIOBool )
             GetSelCHAndAddCH ( pSound->vSelectedInputChannels[i], pSound->lNumInChan,
                                iSelCH, iSelAddCH );
 
+#define CSOUND_GET_DATA(format,endian,factor)                           \

```

Comment:
I actually do not like these types of defines. The resulting code does not contain that much redundancy but the readability of the define as well as the debugging is much worse.

Reply:
I know. Either you repeat that piece of code a dozen times, or you use that macro. I see no simple way to resolve this? Do you?

Reply:
We may not re-invent the wheel but use what is available and well tested. E.g.: https://app.assembla.com/spaces/portaudio/git/source/master/src/hostapi/asio/pa_asio.cpp#ln608

Reply:
I see my changes are in-line with what PortAudio does. I believe my changes are more compact.

Should we add support for PortAudio as a backend?

Reply:
Not with this pull request. There is an Issue for that already: https://github.com/corrados/jamulus/issues/116

> I believe my changes are more compact.

It's not about being more compact, it's more about having a code which is tested and known to work correctly for all bit depths. Verifying the functionality is the most critical part. The most Jamulus users use Windows. The sound interface is critical for stability and reliability of the Jamulus software. That's why I am concerned using almost untested code here.

We should take the portaudio code here and later on we might integrate the portaudio library as an additional option for the sound interface in Jamulus.

Reply:
I'll have a look at it again tonight.